lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -3379,17 +3477,26 @@ Tag('ltx:Math', afterClose => \&cleanup_Math);
 sub cleanup_Math {
   my ($document, $mathnode) = @_;
   # If the Math ONLY contains XMath/XMText, it apparently isn't math at all!?!
-  if (!$document->findnodes('ltx:XMath/ltx:*[local-name() != "XMText"]', $mathnode)) {
+  if (!$document->findnodes('ltx:XMath/ltx:*'
+        . '[local-name() != "XMText"'
+        . ' and local-name() != "XMHint"]', $mathnode)) {
     # So unwrap down to the contents of the XMText's.
-    my @xmtexts = map { $_->childNodes } map { $_->childNodes } $mathnode->childNodes;
-    my @texts   = ();
-    foreach my $text (@xmtexts) {
-      if ($text->nodeType != XML_ELEMENT_NODE) {    # Make sure we've got an element
-        $text = $document->wrapNodes('ltx:text', $text); }
-      $document->addClass($text, 'ltx_markedasmath');   # Now record that it originally was marked as math
-      push(@texts, $text); }
-    $document->replaceNode($mathnode, @texts); }        # and replace the whole Math with the pieces
-  else {                                                # Cleanup any remaining XMTexts
+    my @texts = ();
+    foreach my $child (map { $_->childNodes } $mathnode->childNodes) {
+      if ($document->getNodeQName($child) eq 'ltx:XMHint') {
+        if (my $width = $child->getAttribute('width')) {
+          if (my $space = DimensionToSpaces(Dimension($width))) {
+            push(@texts, $space); } } }
+      else {
+        foreach my $gchild ($child->childNodes) {
+          if ($gchild->nodeType != XML_ELEMENT_NODE) {    # Make sure we've got an element
+            push(@texts, ['ltx:text', { class => 'ltx_markedasmath' }, $gchild]); }
+          else {
+            $document->addClass($gchild, 'ltx_markedasmath');
+            push(@texts, $gchild); } } } }
+    # and replace the whole Math with the pieces
+    $document->replaceTree([undef, undef, @texts], $mathnode); }
```

Comment:
Hmm, that you got it so wrong has me concerned. Text would simply be a string. An undefined element name was intended to "obviously and naturally" imply a fragment. Maybe not so obvious? But then what?

Reply:
Oh so you meant:

```perl
["obviously_this_is_a_fake_node_meant_as_a_fragment", undef, @texts]
```

With the two undefs? The method is already confusing since I'm struggling to visualize what replaces what, and I'll happily support adding a new method that mortal eyes can trivially understand. Maybe something like: 
```perl
  $document->replaceWithNodelist($mathnode, @texts)
```

Reply:
if you have two participants and a replacement, and you can't use either as `self` (which would've been most obvious, as in jquery), a friendly name for me would try to incorporate some ordering information akin to `$document->replaceAwithB($A, $B)`. So `$document->replaceTreeWithFragment($node, $fragment)` clicks immediately.

Reply:
Or if long function names are considered distasteful one could dream up:
```perl
$document->replaceWith(
  old => $node,
  fragments => \@texts);
```

Reply:
I'm a little confused by the confusion. The array representation of an xml node is `[tagname, attributes, children...]`, so undef tagname is a fragment, which doesn't accept attributes, so they're undef too.

But what's more sad is that you overlooked the worse crime which is that this is doing essentially the same as `replaceNode`, but with a different API, and at a higher level (with afterOpen/Close, eg). So, the whole Document API needs a careful going through.